gem/lib/earlgrey/configure_earlgrey_pods.rb

```diff
@@ -0,0 +1,339 @@
+
+        # Add in the Environment Variables
+        launch_action_env_vars.elements.each('EnvironmentVariable') do |launch_action_env_var|
+          environment_variable = REXML::Element.new 'EnvironmentVariable'
+          environment_variable.attributes['key'] = launch_action_env_var.attributes['key']
+          environment_variable.attributes['value'] = launch_action_env_var.attributes['value']
+          environment_variable.attributes['isEnabled'] = launch_action_env_var.attributes['isEnabled']
+          test_action_env_vars.add_element(environment_variable)
+        end
+
+        # Add in the Arguments
+        launch_action_args.elements.each('CommandLineArgument') do |launch_action_arg|
+          argument = REXML::Element.new 'CommandLineArgument'
+          argument.attributes['argument'] = launch_action_arg.attributes['argument']
+          argument.attributes['isEnabled'] = launch_action_arg.attributes['isEnabled']
+          test_action_args.add_element(argument)
+        end
+
+      end
+      scheme.test_action = test_action
+      scheme
+    end
+
+    # Adds EarlGrey.framework to products group. Returns file ref.
+    def add_earlgrey_product
+      return @add_earlgrey_product if @add_earlgrey_product
+      framework_path = if carthage
+                         '${SRCROOT}/Carthage/Build/iOS/EarlGrey.framework'
+                       else
+                         '${SRCROOT}/Pods/EarlGrey/EarlGrey-1.0.0/EarlGrey.framework'
+                       end
+
+      framework_ref = user_project.products_group.files.find { |f| f.path == framework_path }
+      return @add_earlgrey_product = framework_ref if framework_ref
+
+      framework_ref = user_project.products_group.new_file(framework_path)
+      framework_ref.source_tree = 'SRCROOT'
+
+      @add_earlgrey_product = framework_ref
+    end
+
+    # Generates a copy files build phase to embed the EarlGrey framework into
+    # the app under test.
+    def add_earlgrey_copy_files_script
+      earlgrey_copy_files_phase_name = 'EarlGrey Copy Files'
+      earlgrey_copy_files_exists = false
+      test_target.copy_files_build_phases.each do |copy_files_phase|
+        if copy_files_phase.name == earlgrey_copy_files_phase_name
+          earlgrey_copy_files_exists = true
+        end
+      end
+
+      unless earlgrey_copy_files_exists
+        new_copy_files_phase = test_target.new_copy_files_build_phase(earlgrey_copy_files_phase_name)
+        new_copy_files_phase.dst_path = '$(TEST_HOST)/../'
+        new_copy_files_phase.dst_subfolder_spec = '0'
+
+        file_ref = add_earlgrey_product
+        build_file = new_copy_files_phase.add_file_reference(file_ref, true)
+        build_file.settings = { 'ATTRIBUTES' => ['CodeSignOnCopy'] }
+        user_project.save
+      end
+    end
+
+    FRAMEWORK_SEARCH_PATHS = 'FRAMEWORK_SEARCH_PATHS'.freeze
+    HEADER_SEARCH_PATHS = 'HEADER_SEARCH_PATHS'.freeze
+
+    def add_carthage_search_paths
```

Comment:
> Will this add the Carthage paths / Swift code into every project that gets this file, even if it isn't needed?

No.

> We could have a check to enable these latter methods only if they are needed.

There is. That's what `return unless carthage` does.

The gem is setup as a command line tool. By default `--swift` and `--carthage` are true. To opt out, the user would invoke `--no-swift` and `--no-carthage`. That'll trigger the bail out logic in the function making it a no-op.

I'm expecting cocoapods to require the gem directly in the Podfile so that doesn't need a command line option (but it's there anyway in the form of --no-carthage).